packages/opencensus-exporter-ocagent/test/test-ocagent.ts

```diff
@@ -517,4 +519,175 @@ describe('OpenCensus Agent Exporter', () => {
+               });
+               assert.equal(span.attributes.droppedAttributesCount, 0);
+
+               // Time Events
+               assert.deepEqual(span.timeEvents, {
+                 droppedAnnotationsCount: 0,
+                 droppedMessageEventsCount: 0,
+                 timeEvent: [
+                   {
+                     value: 'annotation',
+                     time: null,
+                     annotation: {
+                       description:
+                           {value: 'my_annotation', truncatedByteCount: 0},
+                       attributes: {
+                         attributeMap: {
+                           myString: {
+                             value: 'stringValue',
+                             stringValue:
+                                 {value: 'bar', truncatedByteCount: 0}
+                           },
+                           myNumber: {value: 'intValue', intValue: '123'},
+                           myBoolean: {value: 'boolValue', boolValue: true}
+                         },
+                         droppedAttributesCount: 0
+                       }
+                     }
+                   },
+                   {
+                     messageEvent: {
+                       compressedSize: '0',
+                       id: '65535',
+                       type: 'SENT',
+                       uncompressedSize: '0'
+                     },
+                     time: {seconds: '123456', nanos: 789000000},
+                     value: 'messageEvent'
+                   },
+                   {
+                     value: 'messageEvent',
+                     messageEvent: {
+                       compressedSize: '0',
+                       id: '65535',
+                       type: 'RECEIVED',
+                       uncompressedSize: '0'
+                     },
+                     time: {seconds: '123456', nanos: 789000000},
+                   }
+                 ]
+               });
+
+               // Links
+               const buff = Buffer.from([255, 255]);
+               assert.deepEqual(span.links, {
+                 droppedLinksCount: 0,
+                 link: [
+                   {
+                     type: 'CHILD_LINKED_SPAN',
+                     traceId: buff,
+                     spanId: buff,
+                     attributes: {
+                       droppedAttributesCount: 0,
+                       attributeMap: {
+                         child_link_attribute_string: {
+                           value: 'stringValue',
+                           stringValue: {value: 'foo1', truncatedByteCount: 0}
+                         },
+                         child_link_attribute_number:
+                             {value: 'intValue', intValue: '123'},
+                         child_link_attribute_boolean:
+                             {value: 'boolValue', boolValue: true}
+                       }
+                     }
+                   },
```

Comment:
you have checked the drop counts for links earlier but it would nice to check that oldest links is removed. you can either add this is as part of this or where you test the dropLinkCount.

Reply:
Good Point! We already have asset check here https://github.com/census-instrumentation/opencensus-node/pull/331/files#diff-0ab7878c00c0a6393dd66fd59e796156R482 for the links. Basically, it doesn't contain a dropped link object.

Reply:
as per our conversation, this is already taken care of in previous test 'should adapt a span correctly' on line 320.
so ignore the comment.